$NetBSD: patch-rts_rts.cabal,v 1.1 2025/01/29 13:21:54 pho Exp $

Fix build on Illumos where ld(1) dislikes undefined symbols in executables just like MacOS X.

https://gitlab.haskell.org/ghc/ghc/-/merge_requests/12613

--- rts/rts.cabal.orig	2025-01-09 09:14:35.671178262 +0000
+++ rts/rts.cabal
@@ -335,6 +335,10 @@ library
         if !arch(x86_64) && !arch(aarch64)
            ld-options: -read_only_relocs warning
 
+      if os(solaris2)
+        ld-options: "-Wl,-z,nodefs"
+        -- See Note [Undefined symbols in the RTS]
+
       cmm-sources: Apply.cmm
                    Compact.cmm
                    ContinuationOps.cmm
